app/Models/Signup.php

```diff
@@ -65,21 +65,41 @@ public function posts()
     }
 
     /**
-     * Get the visible posts associated with this signup.
+     * Get the visible posts associated with this signup, optionally by type.
      *
+     * @param string Optional param to specify post type.
      * @return \Illuminate\Database\Eloquent\Relations\HasMany
      */
-    public function visiblePosts()
+    public function visiblePosts($type = null)
     {
         if (! is_staff_user()) {
-            return $this->hasMany(Post::class)->where('status', '=', 'accepted')
-                                              ->orWhere('northstar_id', auth()->id())
-                                              ->with('tags');
+            if ($type === null) {
+                return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+                                                  ->orWhere('northstar_id', auth()->id())
+                                                  ->with('tags');
+            } else {
+                return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+                                                  ->orWhere('northstar_id', auth()->id())
+                                                  ->where('type', '=', $type)
+                                                  ->with('tags');
+            }
         }
 
-        return $this->hasMany(Post::class)->with('tags');
+        if ($type === null) {
+            return $this->hasMany(Post::class)->with('tags');
+        }
+
+
+        if (count($type) > 1) {
+            return $this->hasMany(Post::class)->whereIn('type', array_values($type))
+                                          ->with('tags');
```

Comment:
@DFurnes @katiecrane still need to clean up code but have a quick q! 

I am trying to incorporate @DFurnes 's [great suggestion](https://github.com/DoSomething/rogue/pull/746#issuecomment-401134852) and fidgeting with getting something like this `api/v3/signups?include=posts:type(text|photo)` to work to exclude voter reg posts. Instead of iterating through the params, (in this case, text and photo), I thought that I could use [`whereIn`](https://laravel.com/docs/5.5/queries) instead but am having a hard time getting it to work. 

My question - _should_ `whereIn` be able to work in this case? If so, I'll fidget around more but if not I'll go towards the route of looping through those types !

Reply:
@chloealee I would have expected it to work but that doesn't mean it should haha, looks like in other places we do kind of the same thing with `id` instead of `type`. What does this return that's wrong?

Reply:
Yeah interesting, I would have thought this would work too!

Reply:
It'd be interesting to see what SQL query this is getting compiled to, from [`/__clockwork`](http://rogue.test/__clockwork)

Reply:
@katiecrane it just doesn't return any posts when there are definitely associated `photo` and `text` posts! 

@DFurnes oh that is so neat I've never used that before - thank you! Those queries look very strange to me! Looks like it isn't factoring in `type` at all if I'm reading correctly?
![screen shot 2018-07-05 at 5 07 42 pm](https://user-images.githubusercontent.com/9019452/42348070-0161aed6-8076-11e8-8333-294d6aecdd2a.png)

Reply:
ooo I think I got this to work - by adding the `->get();` at the end!